tests/operator_test.go

```diff
@@ -402,6 +510,173 @@ var _ = Describe("Operator", func() {
 		waitForKv(originalKv)
 		allPodsAreReady(tests.KubeVirtVersionTag)
 
+		if workDir != "" {
+			err = os.RemoveAll(workDir)
+			Expect(err).ToNot(HaveOccurred())
+			workDir = ""
+		}
+
+	})
+
+	Describe("should update kubevirt", func() {
+
+		// This test is installing a previous release of KubeVirt
+		// running a VM/VMI using that previous release
+		// Updating KubeVirt to the target tested code
+		// Ensuring VM/VMI is still operational after the update from previous release.
+		It("from previous release to target tested release", func() {
+			previousImageTag := tests.PreviousReleaseTag
+			previousImageRegistry := tests.PreviousReleaseRegistry
+
+			if previousImageTag == "" {
+				Skip("--previous-release-tag not provided")
+			}
+
+			curTag := originalKv.Status.ObservedKubeVirtVersion
+			curRegistry := originalKv.Status.ObservedKubeVirtRegistry
+
+			allPodsAreReady(tests.KubeVirtVersionTag)
+			sanityCheckDeploymentsExist()
+
+			// Delete current KubeVirt install so we can install previous release.
+			By("Deleting KubeVirt object")
+			deleteAllKvAndWait(false)
+
+			By("Sanity Checking Deployments infrastructure is deleted")
+			sanityCheckDeploymentsDeleted()
+
+			// Install Previous Release of KubeVirt
+			By(fmt.Sprintf("Creating KubeVirt Object with Previous Release: %s using registry %s", previousImageTag, previousImageRegistry))
+			kv := copyOriginalKv()
+			kv.Name = "kubevirt-release-install"
+			kv.Spec.ImageTag = previousImageTag
+			kv.Spec.ImageRegistry = previousImageRegistry
+			createKv(kv)
+
+			// Wait for Previous Release to come online
+			// wait 7 minutes because this test involves pulling containers
+			// over the internet related to the latest kubevirt release
+			By("Waiting for KV to stabilize")
+			waitForKvWithTimeout(kv, 420)
+
+			By("Verifying infrastructure is Ready")
+			allPodsAreReady(previousImageTag)
+			sanityCheckDeploymentsExist()
+
+			// kubectl API discovery cache only refreshes every 10 minutes
+			// Since we're likely dealing with api additions/removals here, we
+			// need to ensure we're using a different cache directory after
+			// the update from the previous release occurs.
+			oldClientCacheDir := workDir + "/oldclient"
+			err = os.MkdirAll(oldClientCacheDir, 0755)
+			Expect(err).ToNot(HaveOccurred())
+			newClientCacheDir := workDir + "/newclient"
+			err = os.MkdirAll(newClientCacheDir, 0755)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Create VM on previous release using a specific API.
+			// NOTE: we are testing with yaml here and explicilty _NOT_ generating
+			// this vm using the latest api code. We want to guarrantee there are no
+			// surprises when it comes to backwards compatiblity with previous
+			// virt apis.  As we progress our api from v1alpha3 -> v1 there
+			// needs to be a VM created for every api. This is how we will ensure
+			// our api remains upgradable and supportable from previous releasei.
```

Comment:
nit "releasei".

Out of curiosity, is the idea to eventually generate a "v1" VM and migrate that one through every step all the way to whatever "now" is? or to start with a fresh one each api version?

Reply:
once the previous release involves multiple apis, it will generate yaml for every api we support as part of the test